fenicsxconcrete/finite_element_problem/linear_elastic_material.py

```diff
@@ -0,0 +1,126 @@
+import warnings
+#from ffc.quadrature.deprecation import QuadratureRepresentationDeprecationWarning
+#df.parameters["form_compiler"]["representation"] = "quadrature"
+#warnings.simplefilter("ignore", QuadratureRepresentationDeprecationWarning)
+
+
+class linear_elasticity(MaterialProblem):
+    """Material definition for linear elasticity"""
+
+    def __init__(self, experiment=None, parameters=None, pv_name='pv_output_linear_elasticity'):
+        """Initializes the object by calling super().__init__
+
+        Parameters
+        ----------
+            experiment : object, optional
+                When no experiment is passed, the dummy experiment "MinimalCubeExperiment" is added
+            parameters : dictionary, optional
+                Dictionary with parameters. When none is provided, default values are used
+            pv_name : string, optional
+                Name of the paraview file, if paraview output is generated
+        """
+        # generate "dummy" experiment when none is passed
+
+        #if experiment is None:
+        #    #experiment = experimental_setups.MinimalCubeExperiment(parameters)
+
+        super().__init__(experiment, parameters, pv_name)
+
+    def setup(self):
+        default_p = Parameters()
+
+        # parameters for mechanics problem
+        default_p['E'] = None  # Young's Modulus
+        default_p['nu'] = None  # Poisson's Ratio
+        default_p['mu'] = None
+        default_p['lmbda'] = None
+
+        self.p = default_p + self.p
+          
+        self.residual = None  # initialize residual
+
+        # Constant E and nu fields.
+        E = self.p.E 
+        nu = self.p.nu 
+
+        self.lambda_ = df.fem.Constant(self.experiment.mesh, E * nu / ((1.0 + nu) * (1.0 - 2.0 * nu)))
+        self.mu = df.fem.Constant(self.experiment.mesh, E / (2.0 * (1.0 + nu)))
+        
+        # Define variational problem
+        self.u_trial = ufl.TrialFunction(self.experiment.V)
+        self.v = ufl.TestFunction(self.experiment.V)
+        
+        # Selects the problem which you want to solve
+        if self.p.problem == 'tensile_test':
+            self.T = df.fem.Constant(self.experiment.mesh, ScalarType((self.p.load, 0)))
+            ds = self.experiment.create_neumann_boundary()
+            self.L =  ufl.dot(self.T, self.v) * ds(1) 
+
+        elif self.p.problem == 'cantilever_beam':
+            if self.p.dim == 2:
+                #f = df.Constant((0, 0))
+                f = df.fem.Constant(self.experiment.mesh, ScalarType((0, -self.p.rho*self.p.g))) 
+            elif self.p.dim == 3:
+                #f = df.Constant((0, 0, 0))
+                f = df.fem.Constant(self.experiment.mesh, ScalarType((0, 0, -self.p.rho*self.p.g))) 
+            else:
+                raise Exception(f'wrong dimension {self.p.dim} for problem setup')
+                
+            self.L =  ufl.dot(f, self.v) * ufl.dx
+        else:
+            exit()
```

Comment:
this is already very problem specific and IMO should not be defined in a class that should cover any linear elastic problem

Reply:
Yes, Erik and I had a discussion about it yesterday. The plan is to move it out from here.